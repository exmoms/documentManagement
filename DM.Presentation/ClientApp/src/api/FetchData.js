/**
 * @file this file includes functions that fetch data from API
 */

// Generic Delete Method
export async function deleteFromApi(url) {
  let res = await fetch(url, {
    method: "DELETE",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  });
  return res;
}

// Get MetaData Models
export async function fetchData(url) {
  let object = {};
  /*
   * send the request through a CORS proxy
   * https://stackoverflow.com/questions/43262121/trying-to-use-fetch-and-pass-in-mode-no-cors/43268098#43268098
   */
  await fetch(url, {
    method: "GET",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  })
    .then((response) => response.json())
    .then(
      (data) => (object = data),
      (error) => console.log(error)
    );
  return object;
}
