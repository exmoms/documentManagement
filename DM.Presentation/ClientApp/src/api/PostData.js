/**
 * @file This file includes functions that help to post data to a specific API
 * @version 1.0
 */

/**
 * Post new document data.
 * Upload form-data, scanned copy of doc and attachments.
 * @author Youssef Shaaban <y.shaaban@lit-co.net>
 * @param {*} documentData
 * @param {*} images
 * @param {*} attachments
 * @param {*} url
 * @param {*} option its values either "Add" for add new document version or "Update" to update exsiting document Version
 */
export async function postDocumentData(
  documentData,
  scannedPages,
  attachments,
  url = "",
  option
) {
  var formData = new FormData();

  if (process.env.NODE_ENV === "development") {
    console.log(documentData);
    console.log(attachments);
    console.log(scannedPages);
  }

  switch (option) {
    case "Add":
      formData.append("Document", JSON.stringify(documentData));
      break;
    case "Update":
      formData.append("Version", JSON.stringify(documentData));
      break;
    default:
  }

  attachments.forEach((attachment) => {
    formData.append("Attachments", attachment);
  });
  scannedPages.forEach((scannedPage) => {
    formData.append("Scans", scannedPage);
  });

  let response = {};
  await fetch(url, {
    // content-type header should not be specified!
    method: "POST",
    headers: {
      Accept: "application/json",
      // "Content-Type": "application/json",
    },
    body: formData,
  }).then(
    (res) => {
      response = res;
      // response.json();
    },
    (error) => console.log("[ERROR]:", error)
  );
  return response;
}

/**
 * Post Attachment Data.
 * Upload attachment json data + attachment file  .
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @param {*} payload
 * @param {*} attachmentFile
 * @param {*} url
 */
export async function postAttachmentData(payload, attachmentFile, url = "") {
  var formData = new FormData();

  formData.append("Attachment", JSON.stringify(payload));
  formData.append("FileAttachment", attachmentFile);

  let response = {};
  await fetch(url, {
    // content-type header should not be specified!
    method: "POST",
    body: formData,
  }).then(
    (res) => {
      response = res;
      // response.json();
    },
    (error) => console.log("[ERROR]:", error)
  );
  return response;
}

/**
 * Post Request To API
 * Send post request to api with  payload  and return a response.
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @param {*} url The api url
 * @param {*} payload js object which will be sent as request body.
 */
export async function postDataToAPI(url, payload = null, method = "POST") {

  const response = await fetch(url, {
    method: method,
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(payload),
  });
  return response;
}
