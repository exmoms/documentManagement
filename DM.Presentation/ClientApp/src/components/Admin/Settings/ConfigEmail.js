/**
 * @file Contains ConfigEmail component implementation
 * @author Youssef Shaaban <y.shaaban@lit-co.net>
 * @version 1.0
 */

import {
  Button,
  Container,
  CssBaseline,
  FormControl,
  IconButton,
  InputAdornment,
  InputLabel,
  OutlinedInput,
  TextField,
} from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import Visibility from "@material-ui/icons/Visibility";
import VisibilityOff from "@material-ui/icons/VisibilityOff";
import clsx from "clsx";
import React, { Component } from "react";
import { withTranslation } from "react-i18next";
import { postDataToAPI } from "../../../api/PostData";
import { fetchData } from "../../../api/FetchData";
import { Notification } from "../../Admin/Notifications";

const useStyles = (theme) => ({
  paper: {
    marginTop: theme.spacing(4),
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  form: {
    width: "100%", // Fix IE 11 issue.
    marginTop: theme.spacing(3),
  },
  submit: {
    margin: theme.spacing(3, 1, 2),
  },
  margin: {
    margin: theme.spacing(1),
  },
  textField: {
    width: "100%",
  },
});

class ConfigEmail extends Component {
  t = this.props["t"];

  addConfigURL = "/api/Mail/AddConfig";
  testMailServerConnectionURL = "/api/Mail/TestConnection";
  getConfigURL = "/api/Mail/GetConfig";
  buttonCase = "";

  constructor(props) {
    super(props);
    this.state = {
      password: "",
      showPassword: false,
      portError: false,
      portErrorText: "",
      testedSuccess: false,
      open: false,
      error: false,
      errorMessage: [],
    };
  }

  componentDidMount() {
    fetchData(this.getConfigURL).then((data) => {
      document.getElementById("email").value = data.email;
      document.getElementById("server").value = data.server;
      document.getElementById("port").value = data.port;
      this.setState({
        password: data.password,
      });
    });
  }

  handlePassword = (e) => {
    this.setState({ password: e.target.value });
  };

  handleClickShowPassword = () => {
    this.setState({ showPassword: !this.state.showPassword });
  };

  handleMouseDownPassword = (event) => {
    event.preventDefault();
  };

  /**
   * handle onChange event for port to check if it's valid
   */
  onChangePort = (e) => {
    const port = e.target.value;

    const validPort =
      port &&
      /^([0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$/i.test(
        port
      );

    if (!validPort) {
      this.setState({
        portError: true,
        portErrorText: this.t("invalid_port"),
      });
    } else {
      this.setState({
        portError: false,
        portErrorText: "",
      });
    }
  };

  handelSubmit = (e) => {
    e.preventDefault();

    const email = document.getElementById("email").value;
    const server = document.getElementById("server").value;
    const port = document.getElementById("port").value;

    const formData = {
      Server: server,
      Port: parseInt(port),
      email: email,
      Password: this.state.password,
    };

    if (
      email !== "" ||
      server !== "" ||
      port !== "" ||
      this.state.password !== ""
    ) {
      switch (this.buttonCase) {
        case "setup":
          if (this.state.testedSuccess) {
            postDataToAPI(this.addConfigURL, formData).then((addRes) => {
              if (addRes.status === 200) {
                this.setState({ open: true, error: false, errorMessage: [] });
                // alert(`Mail configuration is added successfully.`);
              } else {
                this.setState({
                  open: true,
                  error: true,
                  errorMessage: [this.t("Mail_con_err")],
                });
                // alert(`ERROR ${addRes.status}: ${addRes.statusText}`);
              }
            });
          } else {
            // alert("You have to test your configuration and be sure it works.");
            this.setState({
              open: true,
              error: true,
              errorMessage: [this.t("Mail_con_test")],
            });
          }
          break;

        case "test":
          postDataToAPI(this.testMailServerConnectionURL, formData).then(
            (res) => {
              if (res.status === 200) {
                // alert(
                //   `Mail configuration is tested and it successfully works.`
                // );
                this.setState({
                  testedSuccess: true,
                  open: true,
                  error: false,
                });
              } else {
                // alert(`ERROR ${res.status}: ${res.statusText}`);
                this.setState({
                  open: true,
                  error: true,
                  errorMessage: [this.t("Mail_con_test_err")],
                });
              }
            },
            (error) => console.log(error)
          );
          break;

        default:
          console.log(this.buttonCase);
          break;
      }
    }
  };

  setOpen = () => {
    this.setState({ open: false });
  };

  render() {
    const { classes } = this.props;
    const { showPassword, password, portErrorText, portError } = this.state;
    return (
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <div className={classes.paper}>
          <form
            className={classes.form}
            onSubmit={this.handelSubmit}
            id="submitForm"
          >
            <TextField
              className={classes.margin}
              variant="outlined"
              fullWidth
              name="email"
              label={this.t("emailaddress")}
              defaultValue="Enter your email"
              type="email"
              id="email"
            />
            <FormControl
              id="password-form"
              className={clsx(classes.margin, classes.textField)}
              variant="outlined"
            >
              <InputLabel
                htmlFor="outlined-adornment-password"
                id="password-label"
              >
                {this.t("password")}
              </InputLabel>
              <OutlinedInput
                id="OutlinedInput"
                name="password"
                type={showPassword ? "text" : "password"}
                onChange={this.handlePassword}
                value={password}
                endAdornment={
                  <InputAdornment position="end" id="input-adornment-password">
                    <IconButton
                      aria-label="toggle password visibility"
                      onClick={this.handleClickShowPassword}
                      onMouseDown={this.handleMouseDownPassword}
                      edge="end"
                    >
                      {showPassword ? <Visibility /> : <VisibilityOff />}
                    </IconButton>
                  </InputAdornment>
                }
                labelWidth={70}
              />
            </FormControl>
            <TextField
              className={classes.margin}
              variant="outlined"
              fullWidth
              name="server"
              label={this.t("server")}
              defaultValue="Enter server of email"
              type="text"
              id="server"
            />
            <TextField
              className={classes.margin}
              variant="outlined"
              fullWidth
              onChange={this.onChangePort}
              error={portError}
              helperText={portErrorText}
              name="port"
              label={this.t("port")}
              defaultValue={-1}
              type="number"
              id="port"
            />
            <Button
              id="buttonTest"
              type="submit"
              onClick={() => (this.buttonCase = "test")}
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
            >
              {this.t("test")}
            </Button>
            <Button
              id="buttonSetUP"
              type="submit"
              onClick={() => (this.buttonCase = "setup")}
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
            >
              {this.t("setup_email_configuration")}
            </Button>
          </form>
        </div>
        <Notification
          id="notification"
          open={this.state.open}
          setOpen={this.setOpen}
          error={this.state.error}
          errorMessage={this.state.errorMessage}
        />
      </Container>
    );
  }
}

export default withStyles(useStyles)(withTranslation()(ConfigEmail));
