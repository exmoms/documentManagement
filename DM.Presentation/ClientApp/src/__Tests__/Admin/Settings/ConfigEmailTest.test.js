import React from "react";
import ReactDom from "react-dom";
import ConfigEmail from "../../../components/Admin/Settings/ConfigEmail";
import { cleanup } from "@testing-library/react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import TextField from "@material-ui/core/TextField";
import { act } from "react-dom/test-utils";
import Button from "@material-ui/core/Button";
import { BrowserRouter as Router } from "react-router-dom";
import * as FetchMock from "../../../api/FetchData";
import * as PostDataMock from "../../../api/PostData";
import { OutlinedInput, IconButton} from "@material-ui/core";
import Visibility from "@material-ui/icons/Visibility";
import VisibilityOff from "@material-ui/icons/VisibilityOff";
afterEach(cleanup);
describe("ConfigEmail", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  test("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(
        <ConfigEmail />,
      div
    );
  });

  test("matches snapshot", () => {
    const wrapper = shallow(
        <ConfigEmail />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
 
  test("componentDidMount + handelSubmit", () => {
    const preventDefaultMock = jest.fn();
    const mockSuccessResponse = {email:"email" , server:"server", port:"port", password:"password"};
    const mockJsonPromise = Promise.resolve(mockSuccessResponse);
    const fetchMock =  jest.spyOn(FetchMock, "fetchData")
      .mockImplementation(() => mockJsonPromise);
      const mockSuccessResponsePostData = {status:200, json:()=>{return  Promise.resolve( {error:["error"]})}};
      const mockSuccessResponsePostDataNotOk = Promise.resolve({status:300, json:()=>{return  Promise.resolve( {error:["error"]})}});
  const mockJsonPromisePostData = Promise.resolve(mockSuccessResponsePostData); 
  const PostMock=  jest.spyOn(PostDataMock, "postDataToAPI").mockReturnValueOnce(mockJsonPromisePostData).mockReturnValueOnce(mockJsonPromisePostData).mockReturnValueOnce(mockSuccessResponsePostDataNotOk);
      let wrapper;
      return mockJsonPromise.then(() => {
     wrapper = mount(<ConfigEmail />,{ attachTo: document.body }) ;
    }).then(() => {
        wrapper.render();
        wrapper.update();
        //expectation
    expect(fetchMock).toHaveBeenCalledTimes(1);
    expect(fetchMock.mock.calls[0][0]).toBe("/api/Mail/GetConfig");
    expect(wrapper.find(Visibility).length).toBe(0);
    expect(wrapper.find(VisibilityOff).length).toBe(1);
    expect(document.getElementById("email").value).toBe("email");
    expect(document.getElementById("server").value).toBe("server");
    expect(wrapper.find("#port").at(0).props().error).toBeFalsy();
    expect(wrapper.find("#port").at(0).props().helperText).toBe("");
    expect(wrapper.find("#port").at(0).props().type).toBe("number");
    expect(wrapper.find("#OutlinedInput").at(0).props().type).toBe("password");
    expect(wrapper.find("#OutlinedInput").at(0).props().value).toBe("password");
    //expect(document.getElementById("port").value).toBe("port"); //failed test
    //set buttonCase to test
    act(() => {
        wrapper
          .find("#buttonTest")
          .at(0)
          .props()
          .onClick();
      }); 
      wrapper.update();
    return mockJsonPromise.then(() => {
        expect(wrapper.find("#notification").at(0).props().open).toBeFalsy(); 
        act(() => {
            //first branch when buttonCase is "test" and status is 2000
            wrapper
              .find("#submitForm")
              .at(0)
              .props()
              .onSubmit({ preventDefault: preventDefaultMock });
          });
    }).then(() => {
        wrapper.render();
      wrapper.update();
        expect(PostMock).toHaveBeenCalledTimes(1);
        expect(PostMock.mock.calls[0][0]).toBe("/api/Mail/TestConnection");
        expect(PostMock.mock.calls[0][1]).toStrictEqual( {"Password": "password", "Port": NaN, "Server": "server", "email": "email"});
        expect(preventDefaultMock).toHaveBeenCalledTimes(1);
        expect(wrapper.find("#notification").at(0).props().open).toBeTruthy();
        expect(wrapper.find("#notification").at(0).props().error).toBeFalsy();
        //set buttonCase to test
    act(() => {
        wrapper
          .find("#buttonSetUP")
          .at(0)
          .props()
          .onClick();
      }); 
      wrapper.update();
        act(() => {
            //second branch when buttonCase is "setup" and status is 200
            wrapper
              .find("#submitForm")
              .at(0)
              .props()
              .onSubmit({ preventDefault: preventDefaultMock });
          });
          wrapper.render();
         wrapper.update();
        expect(PostMock).toHaveBeenCalledTimes(2);
        expect(PostMock.mock.calls[1][0]).toBe("/api/Mail/AddConfig");
        expect(PostMock.mock.calls[1][1]).toStrictEqual( {"Password": "password", "Port": NaN, "Server": "server", "email": "email"});
        expect(preventDefaultMock).toHaveBeenCalledTimes(2);
        expect(wrapper.find("#notification").at(0).props().open).toBeTruthy();
        expect(wrapper.find("#notification").at(0).props().error).toBeFalsy();
        expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual([]);
        return mockSuccessResponsePostDataNotOk.then(() => {
            act(() => {
                //third branch when buttonCase is "setup" and status isn't 200
                wrapper
                  .find("#submitForm")
                  .at(0)
                  .props()
                  .onSubmit({ preventDefault: preventDefaultMock });
              });
        }).then(() => {
        wrapper.render();
         wrapper.update();
        expect(PostMock).toHaveBeenCalledTimes(3);
        expect(PostMock.mock.calls[2][0]).toBe("/api/Mail/AddConfig");
        expect(PostMock.mock.calls[2][1]).toStrictEqual( {"Password": "password", "Port": NaN, "Server": "server", "email": "email"});
        expect(preventDefaultMock).toHaveBeenCalledTimes(3);
        expect(wrapper.find("#notification").at(0).props().open).toBeTruthy();
        expect(wrapper.find("#notification").at(0).props().error).toBeTruthy();
        expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual(["Mail_con_err"]);
      });
    });
    });

  });

  test("handlePassword", () => {
      let wrapper;
     wrapper = mount(<ConfigEmail />) ;
     expect(wrapper.find("#OutlinedInput").at(0).props().value).toBe("");
    act(() => {
        wrapper
          .find("#OutlinedInput")
          .at(0)
          .props()
          .onChange({ target: { value: "testedVlaue" } });
      }); 
      wrapper.render();
      wrapper.update();
      expect(wrapper.find("#OutlinedInput").at(0).props().value).toBe("testedVlaue");
});

test("handleClickShowPassword", () => {
    let wrapper;
   wrapper = mount(<ConfigEmail />) ;
   expect(wrapper.find(Visibility).length).toBe(0);
    expect(wrapper.find(VisibilityOff).length).toBe(1);
  act(() => {
      wrapper
        .find(IconButton)
        .at(0)
        .props()
        .onClick();
    }); 
    wrapper.render();
    wrapper.update();
    expect(wrapper.find(Visibility).length).toBe(1);
    expect(wrapper.find(VisibilityOff).length).toBe(0);
});

test("handleMouseDownPassword", () => {
    let wrapper;
   wrapper = mount(<ConfigEmail />) ;
   const preventDefaultMock = jest.fn();
  act(() => {
      wrapper
        .find(IconButton)
        .at(0)
        .props()
        .onMouseDown({ preventDefault: preventDefaultMock });
    }); 
    wrapper.render();
    wrapper.update();
    expect(preventDefaultMock).toHaveBeenCalledTimes(1);
});

test("onChangePort", () => {
    let wrapper;
   wrapper = mount(<ConfigEmail />) ;
  act(() => {
      wrapper
        .find("#port")
        .at(0)
        .props()
        .onChange({ target: { value: null } });
    }); 
    wrapper.render();
    wrapper.update();
    expect(wrapper.find("#port").at(0).props().error).toBeTruthy();
    expect(wrapper.find("#port").at(0).props().helperText).toBe("invalid_port");
    act(() => {
        wrapper
          .find("#port")
          .at(0)
          .props()
          .onChange({ target: { value: 123 } });
      }); 
      wrapper.render();
      wrapper.update();
      expect(wrapper.find("#port").at(0).props().error).toBeFalsy();
      expect(wrapper.find("#port").at(0).props().helperText).toBe("");
});
test("handelSubmit not ok branches", () => {
    const preventDefaultMock = jest.fn();
    const mockSuccessResponse = {email:"email" , server:"server", port:"port", password:"password"};
    const mockJsonPromise = Promise.resolve(mockSuccessResponse);
    const fetchMock =  jest.spyOn(FetchMock, "fetchData")
      .mockImplementation(() => mockJsonPromise);
      const mockSuccessResponsePostDataNotOk = Promise.resolve({status:300, json:()=>{return  Promise.resolve( {error:["error"]})}});
  const PostMock=  jest.spyOn(PostDataMock, "postDataToAPI").mockReturnValueOnce(mockSuccessResponsePostDataNotOk);
      let wrapper;
      return mockJsonPromise.then(() => {
     wrapper = mount(<ConfigEmail />) ;
    }).then(() => {
        wrapper.render();
        wrapper.update();
        //expectation
    expect(fetchMock).toHaveBeenCalledTimes(1);
    expect(fetchMock.mock.calls[0][0]).toBe("/api/Mail/GetConfig");
    //set buttonCase to test
    act(() => {
        wrapper
          .find("#buttonTest")
          .at(0)
          .props()
          .onClick();
      }); 
      wrapper.update();
    return mockSuccessResponsePostDataNotOk.then(() => {
        expect(wrapper.find("#notification").at(0).props().open).toBeFalsy(); 
        expect(wrapper.find("#notification").at(0).props().error).toBeFalsy(); 
        expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual([]);
        act(() => {
            //when testedSuccess is not success
            wrapper
              .find("#submitForm")
              .at(0)
              .props()
              .onSubmit({ preventDefault: preventDefaultMock });
          });
    }).then(() => {
        wrapper.render();
      wrapper.update();
        expect(PostMock).toHaveBeenCalledTimes(1);
        expect(PostMock.mock.calls[0][0]).toBe("/api/Mail/TestConnection");
        expect(PostMock.mock.calls[0][1]).toStrictEqual( {"Password": "password", "Port": NaN, "Server": "server", "email": "email"});
        expect(preventDefaultMock).toHaveBeenCalledTimes(1);
        expect(wrapper.find("#notification").at(0).props().open).toBeTruthy();
        expect(wrapper.find("#notification").at(0).props().error).toBeTruthy();
        expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual(["Mail_con_test_err"]);
       //set buttonCase to test
    act(() => {
        wrapper
          .find("#buttonSetUP")
          .at(0)
          .props()
          .onClick();
      }); 
      wrapper.update();

      act(() => {
        //when testedSuccess is false
        wrapper
          .find("#submitForm")
          .at(0)
          .props()
          .onSubmit({ preventDefault: preventDefaultMock });
      });
      wrapper.render();
      wrapper.update();
      expect(PostMock).toHaveBeenCalledTimes(1);
      expect(preventDefaultMock).toHaveBeenCalledTimes(2);
      expect(wrapper.find("#notification").at(0).props().open).toBeTruthy();
        expect(wrapper.find("#notification").at(0).props().error).toBeTruthy();
        expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual(["Mail_con_test"]);
        //calling setOpen 
        act(() => {
            wrapper
              .find("#notification")
              .at(0)
              .props()
              .setOpen();
          });
          wrapper.render();
          wrapper.update();
          expect(wrapper.find("#notification").at(0).props().open).toBeFalsy();
    });
    });
});
test("handelSubmit when default case", () => {
    let wrapper;
   wrapper = mount(<ConfigEmail />) ;
   const preventDefaultMock = jest.fn();
   const logMock =  jest.spyOn(console, "log")
   act(() => {
    //when testedSuccess is false
    wrapper
      .find("#submitForm")
      .at(0)
      .props()
      .onSubmit({ preventDefault: preventDefaultMock });
  });
  wrapper.render();
  wrapper.update();
    expect(preventDefaultMock).toHaveBeenCalledTimes(1);
    expect(logMock).toHaveBeenCalledTimes(1);
    expect(logMock.mock.calls[0][0]).toBe("");
});
});