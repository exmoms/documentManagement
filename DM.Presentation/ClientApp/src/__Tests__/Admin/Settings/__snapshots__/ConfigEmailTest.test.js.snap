// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigEmail matches snapshot 1`] = `
<withI18nextTranslation(ConfigEmail)
  classes={
    Object {
      "form": "withI18nextTranslation(ConfigEmail)-form-2",
      "margin": "withI18nextTranslation(ConfigEmail)-margin-4",
      "paper": "withI18nextTranslation(ConfigEmail)-paper-1",
      "submit": "withI18nextTranslation(ConfigEmail)-submit-3",
      "textField": "withI18nextTranslation(ConfigEmail)-textField-5",
    }
  }
/>
`;
